versionutils/repo.go

```diff
@@ -155,3 +157,23 @@ func ParseFullTomlFromDir(relativeDir string) (*TomlWrapper, error) {
 func ParseFullToml() (*TomlWrapper, error) {
 	return ParseFullTomlFromDir("")
 }
+
+func getVersionFromTree(tomlTree *toml.Tree, pkgName string) (version string, found bool) {
+	isEmpty := func(node *toml.Tree, key string) bool {
+		return node.Get(key) == nil || node.Get(key) == ""
+	}
+
+	if tomlTree.Get(nameConst) != pkgName {
+		return
+	}
+
+	switch {
+	case !isEmpty(tomlTree, versionConst):
+		return tomlTree.Get(versionConst).(string), true
+	case !isEmpty(tomlTree, revisionConst):
+		return tomlTree.Get(revisionConst).(string), true
+	case !isEmpty(tomlTree, branchConst):
+		return tomlTree.Get(branchConst).(string), true
+	}
+	return
```

Comment:
the naked return is only supposed to be used in super short functions (~3-5 lines of code)

i think we should explicitly return `"", false` here